docs/source/training.mdx

```diff
@@ -10,385 +10,351 @@ an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express o
 
-To check if this works on practice, let's create a new [`Trainer`] with our fine-tuned model:
+```py
+>>> from transformers import TrainingArguments
 
-```python
-trainer = Trainer(
-    model=model,
-    args=training_args,
-    train_dataset=small_train_dataset,
-    eval_dataset=small_eval_dataset,
-    compute_metrics=compute_metrics,
-)
-trainer.evaluate()
+>>> training_args = TrainingArguments(output_dir="test_trainer", evaluation_strategy="epoch")
 ```
 
-which showed an accuracy of 87.5% in our case.
+### Trainer
 
-If you want to fine-tune your model and regularly report the evaluation metrics (for instance at the end of each
-epoch), here is how you should define your training arguments:
+Create a [`Trainer`] with your model, training arguments, training and test datasets, and evaluation function:
 
-```python
-from transformers import TrainingArguments
-
-training_args = TrainingArguments("test_trainer", evaluation_strategy="epoch")
+```py
+>>> trainer = Trainer(
+...     model=model,
+...     args=training_args,
+...     train_dataset=small_train_dataset,
+...     eval_dataset=small_eval_dataset,
+...     compute_metrics=compute_metrics,
+... )
 ```
 
-See the documentation of [`TrainingArguments`] for more options.
+Then fine-tune your model by calling `train()`:
 
+```py
+>>> trainer.train()
+```
 
 <a id='keras'></a>
 
-## Fine-tuning with Keras
+## Fine-tune with Keras
 
 <Youtube id="rnTGBy2ax1c"/>
 
-Models can also be trained natively in TensorFlow using the Keras API. First, let's define our model:
+🤗 Transformers models can also be trained natively in TensorFlow with the Keras API. You only need to make a few changes before you can fine-tune.
 
-```python
-import tensorflow as tf
-from transformers import TFAutoModelForSequenceClassification
+### Convert dataset to TensorFlow format
 
-model = TFAutoModelForSequenceClassification.from_pretrained("bert-base-cased", num_labels=2)
-```
+First, you need a [`DataCollator`](main_classes/data_collator) to assemble your tensors into a batch. You can use [`transformers.DataCollatorWithPadding`] to also apply a padding strategy. This will pad each input to the longest length in each batch (also known as _dynamic padding_), which will speed up your training. Make sure you specify `return_tensors` to return TensorFlow tensors:
 
-Then we will need to convert our datasets from before in standard `tf.data.Dataset`. Since we have fixed shapes,
-it can easily be done like this. First we remove the _"text"_ column from our datasets and set them in TensorFlow
-format:
+```py
+>>> from transformers import DataCollatorWithPadding
 
-```python
-tf_train_dataset = small_train_dataset.remove_columns(["text"]).with_format("tensorflow")
-tf_eval_dataset = small_eval_dataset.remove_columns(["text"]).with_format("tensorflow")
+>>> data_collator = DataCollatorWithPadding(tokenizer=tokenizer, return_tensors="tf")
```

Comment:
We are not presenting this on the PyTorch side, and the dataset has been padded to the maximum length. So let's ignore the data collator in this tutorial and write another tutorial where we do the dynamic padding for both?

Reply:
> So let's ignore the data collator in this tutorial

I agree, but we run into a little problem with `to_tf_dataset()` which requires the `collate_fn` parameter. So I think we need to introduce a data collator, but maybe we can use `DefaultDataCollator` instead of `DataCollatorWithPadding`? Sorry if this is what you meant, and I misunderstood! 😅

Reply:
@Rocketknight1 Why does the function need a data collator when everything is rectangular?

Reply:
Hi, you're right that we could just use the default data collator for this! We do need a data collator of some kind, though, because they handle things like renaming label columns.

Reply:
Thanks, updated to use `DefaultDataCollator` instead!